src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Stream.cs

```diff
@@ -183,7 +183,7 @@ public static partial class JsonSerializer
                     else if (bytesInBuffer != 0)
                     {
                         // Shift the processed bytes to the beginning of buffer to make more room.
-                        Buffer.BlockCopy(buffer, bytesConsumed, buffer, 0, bytesInBuffer);
+                        Buffer.BlockCopy(buffer, bytesConsumed + start, buffer, 0, bytesInBuffer);
```

Comment:
Note the similar LOC above was already correct: https://github.com/dotnet/corefx/blob/1fe4d5b844287f6f68926dc44749dd23f588ba67/src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Stream.cs#L174-L175